packages/fxa-settings/src/lib/context/interfaces/context-provider.ts

```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Context } from "./context";
+
+export interface ContextProvider extends Record<string, any>
```

Comment:
It feels a little funny naming these `context` / `context-provider` to me since I think it may cause some confusion when creating or referencing React's context + providers but I'm not sure of better naming off the top of my head except maybe `model` for `context-provider`...

Reply:
Lol, good call out. This name keeps confusing VSCODE auto imports. I'll come up with something better.

Reply:
Okay, I am thinking make this more explicit. 

`context-provider -> BoundModel` _(It has a validate method so is more than a pure provider now anyways.)_

`context -> ModelContext` _(Which still conflicts a little bit, but I think at least pretty explicit)_